apps/coscrad-frontend/src/utils/generic-components/presenters/generic-index-presenter.tsx

```diff
@@ -9,11 +10,26 @@ export const GenericIndexPresenter: FunctionalComponent<IIndexQueryResult<IBaseV
 
     return (
         <div>
-            {viewModels.map((viewModel) => (
-                <Link to={viewModel.id} key={viewModel.id}>
-                    <div data-testid={viewModel.id}>{JSON.stringify(viewModel)}</div>
-                </Link>
-            ))}
+            <h3>Records</h3>
+            <div className="records-table">
+                <table border={1} cellSpacing={0}>
+                    <tbody>
+                        <tr>
+                            <th>Link</th>
+                            {Object.keys(viewModels[0]).map((propertyName) => (
+                                <th>{propertyName}</th>
+                            ))}
+                        </tr>
+                        {viewModels.map((viewModel) => (
+                            <GenericTableRowPresenter key={viewModel.id} {...viewModel} />
+                        ))}
+                    </tbody>
+                </table>
+            </div>
+            <h3>JSON Data</h3>
+            <div className="json-data">
+                <pre>{JSON.stringify(viewModels, null, 2)}</pre>
```

Comment:
We may want to have a util for serializing view models so we can tweak the strategy (JSON.stringify right now) without touching this file.

Reply:
We're going to remove this piece and we have such a utility in the generic index table dir, so no worries.